docs/src/main/sphinx/connector/pushdown-char-varchar-case.fragment

```diff
@@ -0,0 +1,16 @@
+Predicate pushdown support
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector does not push down predicates (such as
+``WHERE column_name = 'value'``) on a ``CHAR`` or ``VARCHAR`` type column unless
+the column name's case exactly matches the column name in the data source. For
+example, the predicate in the following query is only pushed down if the table
+uses "CountryName" exactly as the column name.
```

Comment:
It's not about the case of column-name. It's about the "collation" that the remote system is using. In short collations define how values of text types are compared.

1. They can be case-sensitive (such that `a != A`) or case-insensitive (such that `a = A`).
2. Two case-sensitive collations may sort values differently too (e.g. PostgreSQL's 'en_US.utf8` sorts `a A b B` while Trino/Oracle does `A B a b`. PostgreSQL's `C` (or `POSIX`) collation on the other hand sorts same as Trino.

Now, for what we intend to document here I think we need something like:

For ClickHouse, MemSQL, MySQL and SQL Server:
```rst
This connector does not push down any predicates on columns with textual types like ``CHAR`` or ``VARCHAR``. This is to ensure correctness of results since <MemSQL/MySQL/what-have-you> compare strings case-insensitively by-default.

.. code-block:: sql

    SELECT * FROM nation WHERE name > 'CANADA';
    SELECT * FROM nation WHERE name = 'CANADA';

In this example the predicate is not pushed down for either query since ``name`` is a column of type ``VARCHAR``.
```

For PostgreSQL (unless https://github.com/trinodb/trino/pull/9746 gets merged in first):
```rst
This connector does not push down inequality predicates (like `!=`, `>`, `<` etc.) on columns with textual types like ``CHAR`` or ``VARCHAR``.  Equality predicates (like `IN`, `=` etc.) on columns with textual types are pushed down. This is to ensure correctness of results since <PostgreSQL/etc> sort strings differently than Trino by-default.

.. code-block:: sql

    SELECT * FROM nation WHERE name > 'CANADA';
    SELECT * FROM nation WHERE name = 'CANADA';

In this example for the first query the predicate is not pushed down since ``name`` is a column of type ``VARCHAR`` and an inequality predicate (`>`) is applied while for the second query the predicate is pushded down since an equality predicate (`=`) is applied.
```

Also it might be better inline this limitation instead of a fragment since it's possible each connector might end up with different behaviour for now (on the path to eventual convergence).

Reply:
Perfect, thanks for the clarification - will make these changes shortly.

For now I'd prefer to take the fragmentation approach since it's currently content that can be imported easily. It shouldn't be difficult for us to follow-up later with more bespoke wording if needed.